hibernate-core/src/main/java/org/hibernate/generator/values/GeneratedValueBasicResultBuilder.java

```diff
@@ -103,19 +103,12 @@ public BasicValuedModelPart getModelPart() {
 	}
 
 	private static int columnIndex(JdbcValuesMetadata jdbcResultsMetadata, BasicValuedModelPart modelPart) {
-		try {
-			return jdbcPositionToValuesArrayPosition( jdbcResultsMetadata.resolveColumnPosition(
-					getActualGeneratedModelPart( modelPart ).getSelectionExpression()
-			) );
-		}
-		catch (Exception e) {
-			if ( modelPart.isEntityIdentifierMapping() ) {
-				// Default to the first position for entity identifiers
-				return 0;
-			}
-			else {
-				throw e;
-			}
+		if ( modelPart.isEntityIdentifierMapping() ) {
+			// Default to the first position for entity identifiers
+			return 0;
```

Comment:
I'm not sure this is safe, what if an identifier mapping is present in the result set but is not in the first position? Maybe for dialects that don't support returning arbitrary generated values it's fine, but there might be `n` columns here and if we call `columnIndex` it's because we can't ensure their order/position.

Reply:
Hibernate always and people used to create table using id column as first column, I think it's safe enough to assume that.
Anyway, this commit doesn't overturn the assumption you previous made, So It improved something but doesn't break anything.

Reply:
We should ensure id column is the first of `columnNames` for `inferredKeys=true`, I'm not sure it's already done by Hibernate.

Reply:
I think it might be at the moment, but I thought of a simpler solution: https://github.com/hibernate/hibernate-orm/pull/8391

For dialects not supporting `getGeneratedKeys` with custom column names the result set will always only contain 1 column: the identifier. This way we still leave the possibility of calling `columnIndex` on an arbitrarily-ordered result set, without _needing_ to ensure consistent id positions.

Reply:
> For dialects not supporting `getGeneratedKeys` with custom column names the result set will always only contain 1 column: the identifier. This way we still leave the possibility of calling `columnIndex` on an arbitrarily-ordered result set, without _needing_ to ensure consistent id positions.

Yes, It's reasonable.